iModelCore/ecobjects/src/SchemaMerger.cpp

```diff
@@ -343,6 +390,13 @@ BentleyStatus SchemaMerger::MergeItems(SchemaMergeResult& result, ECSchemaP left
         if (opCode == ECChange::OpCode::New)
             { //This may actually work
             auto newSchemaItem = (right->*getItemCP)(itemName);
+            if (newSchemaItem == nullptr)
+                {
+                result.Issues().ReportV(IssueSeverity::Fatal, IssueCategory::BusinessProperties, IssueType::ECSchema, ECIssueId::EC_0059,
+                    "The schema is dirty. Another reference of the item with name %s already exists in the schema.", right->GetFullSchemaName().c_str());

```

Comment:
I do not think callers would understand this log message.
I suggest something a long the lines of "Failed to find item with name %s in right schema with name %s. This usually indicates a dirty schema graph where multiple memory references of the same schema with different contents are provided"